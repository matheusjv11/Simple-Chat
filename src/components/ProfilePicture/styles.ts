import styled, { DefaultTheme, css } from 'styled-components'
import { ProfilePictureProps } from '.'

type WrapperProps = Pick<ProfilePictureProps, 'size'> & { profile: string }

const imageSizeModifiers = {
  small: (theme: DefaultTheme) => css`
    height: ${theme.sizes.smallImageSize};
    width: ${theme.sizes.smallImageSize};
    min-width: ${theme.sizes.smallImageSize};
  `,

  normal: (theme: DefaultTheme) => css`
    height: ${theme.sizes.imageSize};
    width: ${theme.sizes.imageSize};
    min-width: ${theme.sizes.imageSize};
  `,

  large: (theme: DefaultTheme) => css`
    height: 100%;
    width: 100%;
    border-radius: 0;
  `
}

export const Wrapper = styled.main<WrapperProps>`
  ${({ theme, profile, size }) => css`
    background-image: url(${profile});
    background-position: center;
    background-size: cover;
    border-radius: 100%;
    position: relative;
    box-shadow: rgba(255, 255, 255, 0.1) 0px 1px 1px 0px inset,
      rgba(50, 50, 93, 0.25) 0px 50px 100px -20px,
      rgba(0, 0, 0, 0.3) 0px 30px 60px -30px;

    ${!!size && imageSizeModifiers[size](theme)}
  `}
`

export const OnlineStatus = styled.div`
  background-color: #88cba6;
  height: 13px;
  width: 13px;
  border-radius: 100%;
  position: absolute;
  bottom: -1px;
  right: 3px;
  border: 1px solid white;
`
