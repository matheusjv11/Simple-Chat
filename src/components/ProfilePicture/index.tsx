import * as S from './styles'

export type ProfilePictureProps = {
  profile: string
  profileAlt: string
  online: boolean
  size?: 'small' | 'normal'
}

const ProfilePicture = ({
  profile,
  online,
  profileAlt,
  size = 'normal'
}: ProfilePictureProps) => (
  <S.Wrapper
    profile={profile}
    title={profileAlt}
    aria-label={profileAlt}
    size={size}
  >
    {online && <S.OnlineStatus />}
  </S.Wrapper>
)

export default ProfilePicture
