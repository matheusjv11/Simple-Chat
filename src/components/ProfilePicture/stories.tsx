import type { Meta, StoryObj } from '@storybook/react';

import ProfilePicture from '.';

const meta: Meta<typeof ProfilePicture> = {
  component: ProfilePicture,
};

export default meta;

type Story = StoryObj<typeof ProfilePicture>;

export const SingleUser: Story = {
  render: () => <ProfilePicture />,
};