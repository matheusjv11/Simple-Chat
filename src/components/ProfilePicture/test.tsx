import { render, screen } from '@testing-library/react'

import ProfilePicture from '.'

describe('<ProfilePicture />', () => {
  it('should render the heading', () => {
    const { container } = render(<ProfilePicture />)
    
    expect(
      screen.getByRole('heading', { name: /ProfilePicture/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})