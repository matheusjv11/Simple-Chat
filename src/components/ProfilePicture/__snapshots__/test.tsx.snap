// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePicture /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    ProfilePicture
  </h1>
</main>
`;
